processor/src/lib.rs

```diff
@@ -443,4 +443,43 @@ where
             self.advice_provider,
         )
     }
+
+    // INTERNALS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the process system.
+    #[cfg(feature = "internals")]
+    pub fn system(&self) -> &System {
+        &self.system
+    }
+
+    /// Returns the process decoder.
+    #[cfg(feature = "internals")]
+    pub fn decoder(&self) -> &Decoder {
+        &self.decoder
+    }
+
+    /// Returns the process stack.
+    #[cfg(feature = "internals")]
+    pub fn stack(&self) -> &Stack {
```

Comment:
ATM this is the only method used by the tests on `miden-base`. I see no harm in adding the other methods behind the flag though.